db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2628,9 +2643,12 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
     status = versions_->LogAndApply(c->column_family_data(),
                                     *c->mutable_cf_options(), c->edit(),
                                     &mutex_, directories_.GetDbDir());
+    auto callback = std::bind(&DBImpl::NotifyOnCompactionCompleted, this,
+        c->column_family_data(), c.get(), status, compaction_job_stats,
+        job_context->job_id);
     InstallSuperVersionAndScheduleWork(c->column_family_data(),
                                        &job_context->superversion_contexts[0],
-                                       *c->mutable_cf_options());
+                                       *c->mutable_cf_options(), callback);
```

Comment:
Just a side note that has nothing to do with the PR per se but we might have a small bug here: `InstallSuperVersionAndScheduleWork` is called here (and on the trivial move branch) regardless of `status`.

Reply:
I have noticed that as well. Also there is a different behavior between NotifyOnCompactionCompleted and NotifyOnFlushCompleted. NotifyOnFlushCompleted only triggers when status.ok(), but NotifyOnCompactionCompleted triggers regardless of status. I kept the existing logic as it is, and we can have separate PR to address them.

Reply:
Yeah, this is definitely out of scope here. Also, IIRC `FlushJobInfo` does not even have a `status` field.